samples/configureRetries.js

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const {
+  IdempotencyStrategy,
+} = require('@google-cloud/storage/build/src/storage');
+
+/**
+ * This application demonstrates how to perform basic operations on buckets with
+ * the Google Cloud Storage API.
+ *
+ * For more information, see the README.md under /storage and the documentation
+ * at https://cloud.google.com/storage/docs.
+ */
+
+function main(bucketName = 'my-bucket', fileName = 'test.txt') {
+  // [START storage_configure_retries]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // The ID of your GCS bucket
+  // const bucketName = 'your-unique-bucket-name';
+
+  // The ID of your GCS file
+  // const fileName = 'your-file-name';
+
+  // Imports the Google Cloud client library
+  const {Storage} = require('@google-cloud/storage');
+
+  // Creates a client
+  const storage = new Storage({
+    retryOptions: {
+      // If this is false, requests will not retry and the parameters 
+      // below will not affect retry behavior.
+      autoRetry: true, 
+      // The multiplier by which to increase the delay time between the 
+      // completion of failed requests, and the initiation of the subsequent 
+      // retrying request.
+      retryDelayMultiplier: 3,
+      // The total time between an initial request getting sent and its timeout. 
+      // After timeout, an error will be returned regardless of any retry attempts 
+      // made during this time period.
+      totalTimeout: 500,
```

Comment:
LGTM! Just a question here,  in this case, `maxRetries: 5` would take precedence of `totalTimeout`, correct?

Reply:
what do you mean by that? once the timeout is hit, it will stop retrying, regardless of the number of retries completed

Reply:
Right, but in this case if we do the calculation, 5 retries will only take around 100 seconds. Once the `maxRetries: 5` is exhausted, it will stop retrying, regardless of the `totalTimeout`?

Reply:
probably correct. totalTimeout is not just the additive time of all the retries, but also includes the time the function takes to run. so if it's a long running function, all 5 might not be exhausted.

Reply:
Ahhhh I see, that makes sense. Thanks for the clarification!